src/Extension.Tests/ChallengeTests.cs

```diff
@@ -86,18 +75,15 @@ public async Task challenge_tracks_events_in_submission_history()
         {
             var capturedEvents = new List<List<KernelEvent>>();
             var lesson = new Lesson();
-            using var kernel = new CompositeKernel
-            {
-                new CSharpKernel()
-            }.UseLessonEvaluateMiddleware(lesson);
+            using var kernel = CreateKernel(lesson);
             var challenge = GetEmptyChallenge();
             await lesson.StartChallengeAsync(challenge);
```

Comment:
same here, starting a challenge that is not part of a lesson should not be allowed